Userland/Applications/PixelPaint/PaletteWidget.cpp

```diff
@@ -162,4 +136,87 @@ void PaletteWidget::set_secondary_color(Color color)
     m_secondary_color_widget->update();
 }
 
+Result<void, String> PaletteWidget::load_palette_file(String const& file_path)
+{
+    auto file_or_error = Core::File::open(file_path, Core::OpenMode::ReadOnly);
+    if (file_or_error.is_error())
+        return file_or_error.error();
+
+    auto& file = *file_or_error.value();
+
+    Vector<Color> new_colors;
+
+    while (file.can_read_line()) {
+        auto line = file.read_line();
+        auto color = Color::from_string(line);
+        if (!color.has_value()) {
+            return String::formatted("Could not parse \"{}\" as a color", line);
+        }
```

Comment:
Should this be a hard error? I feel like we caan keep parsing the palette, and only return a hard error if no color matches at all.

In any case, I think we can always ignore some cases, e.g. empty lines.

Reply:
I'll add a check for `line.is_whitespace()` so we just ignore empty-ish lines, and make `load_palette_file` return an error in case not a single color could be parsed from the file.